rfcs/formats/views/request.md

```diff
@@ -8,7 +8,92 @@
 - RFC: 03-007
 - Authors: Venkatraman Ramakrishna, Sandeep Nishad, Krishnasuri Narayanam, Dhinakaran Vinayagamurthy
 - Status: Proposed
-- Since: 02-Apr-2022
+- Since: 30-May-2022
 
 ## Summary
 
+## Query
+
+Query message is used to define the data that is being requested. This is used in relay-relay and relay-driver communication, where destination relay requests response for this query from source relay, and source relay requests response from the driver identified by the [view address](../../formats/views/addressing.md) provided in this payload. This is different from [NetworkQuery](../communication/relay.md#networkquery) as that message is used by client application at destination to initiate the data sharing query request to the destination relay. This message additionally contains `request_id` generated by the destination relay to uniquely identify and track this query request.
+
+```protobuf
+message Query {
+  repeated string policy = 1;
+  string address = 2;
+  string requesting_relay = 3;
+  string requesting_network = 4;
+  string certificate = 5;
+  string requestor_signature = 6;
+  string nonce = 7;
+  string request_id = 8;
+  string requesting_org = 9;
+  bool confidential = 10;
+}
+```
+
+## State
+
+### ViewPayload
+
+Encapsulates either error or the response view of the query request identified by `request_id`. The driver uses this payload to encapsulate the response view or error (depending upon the query response from the ledger) into `ViewPayload` message, and send it to the source relay using the [SendDriverState](../../models/infrastructure/relays.md#api-for-driver) API. Source relay will send the same payload to the destination relay using [SendState](../../models/infrastructure/relays.md#api-for-other-relays) API. The destination relay then stores the state in the database indexed by `request_id`. 
+
+```protobuf
+// View represents the response from a remote network
+message ViewPayload {
+  string request_id = 1;
+  oneof state {
+    View view = 2;
+    string error = 3;
+  };
+}
+```
+
+The `state` either contains `view` field, an instance of `View` defined [here](./view-definition.md) (if the view/data sharing request was successful), otherwise it consists of an error report.
```

Comment:
Rephrase to: The `state` either contains a `view` field (which is an instance of [View](<link>), if the view/data sharing request was successful), otherwise it consists of an error report.